sys/net/gnrc/application_layer/dhcpv6/client.c

```diff
@@ -174,6 +174,17 @@ void dhcpv6_client_conf_prefix(unsigned iface, const ipv6_addr_t *pfx,
             gnrc_rpl_root_init(CONFIG_GNRC_RPL_DEFAULT_INSTANCE, &addr, false, false);
         }
     }
+
+    /* start advertising subnet obtained via DHCPv6 */
+    gnrc_ipv6_nib_change_rtr_adv_iface(netif, true);
```

Comment:
Same here, why make this a general thing and not config-specific?

Reply:
In which case would one not want the prefix to be advertised on the downstream interface?
This is already the behavior in master, the only change is that RAs are disabled until the DHCPv6 configuration has been received.

Reply:
But why? Shouldn't a router advertise themselves as such? Even if there is no DHCPv6 configuration yet?

Reply:
Without the DHCPv6 configuration, the router can't do any routing yet.
By sending the RA right after we received the configuration we can ensure that the downstream nodes get configured as fast as possible.

Keeping RAs disabled at boot until a configuration has been received simplifies the logic.

Reply:
Same argument as in https://github.com/RIOT-OS/RIOT/pull/16590#discussion_r666120328 applies.

> Without the DHCPv6 configuration, the router can't do any routing yet.

If there is a routing protocol running on the border router then it can do. Just because it isn't the case in the current use case, does not mean it isn't possible. I am not sure, but I believe, given that it introduces a router of higher authority than the BR, even https://datatracker.ietf.org/doc/html/rfc8505 would make routing possible without any DHCPv6.